pkg/cache/cache.go

```diff
@@ -658,6 +663,7 @@ func (c *Cache) ForgetWorkload(w *kueue.Workload) error {
 		return ErrCqNotFound
 	}
 	cq.deleteWorkload(w)
+	delete(cq.workloadsNotAccountedForTAS, workload.Key(w))
```

Comment:
Worth moving this into `deleteWorkload` there are other places where we should also delete this from the map